Lib/poplib.py

```diff
@@ -217,6 +218,51 @@ def pass_(self, pswd):
         """
         return self._shortcmd('PASS %s' % pswd)
 
+    def auth(self, mechanism, authobject=None, initial_response=None):
+        """Authenticate to the POP3 server using the AUTH command (RFC 5034).
+
+        Result is 'response'.
+        """
+        if authobject is not None and initial_response is not None:
+            raise ValueError('authobject and initial_response are mutually exclusive')
+
+        if initial_response is not None:
+            if isinstance(initial_response, str):
+                initial_response = initial_response.encode(self.encoding)
+            b64 = base64.b64encode(initial_response).decode('ascii')
+            return self._shortcmd(f'AUTH {mechanism} {b64}'.rstrip())
+
+        if authobject is None:
+            return self._shortcmd(f'AUTH {mechanism}')
+
+        self._putcmd(f'AUTH {mechanism}')
+        while True:
+            resp = self._getresp()
+            if resp[:3] == b'+OK':
+                return resp
+
+            challenge_b64 = resp[1:].lstrip(b' ')
+            if challenge_b64:
+                try:
+                    challenge = base64.b64decode(challenge_b64)
+                except Exception:
+                    padded = challenge_b64 + b'=' * (-len(challenge_b64) % 4)
+                    challenge = base64.b64decode(padded, validate=False)
```

Comment:
We need to check for decode failure here, as well, and if it fails, the RFC requires we send back a '*' response to cancel the auth.  And then raise an -ERR, I think.